pyop2/codegen/rep2loopy.py

```diff
@@ -655,12 +649,21 @@ def statement_functioncall(expr, context):
 
     call = pym.Call(pym.Variable(expr.name), tuple(reads))
 
-    return loopy.CallInstruction(tuple(writes), call,
+    insn = loopy.CallInstruction(tuple(writes), call,
                                  within_inames=within_inames,
                                  predicates=predicates,
                                  id=id,
                                  depends_on=depends_on, depends_on_is_final=True)
 
+    # We cannot reconstruct the correct calling convention for C-string kernels
+    # without providing some additional context about the argument ordering.
+    # This is processed inside the ``emit_call_insn`` method of
+    # :class:`.PyOP2KernelCallable`.
+    if not hasattr(context, "function_args"):
+        context.function_args = {}
+    context.function_args[id] = function_args
```

Comment:
Do we guarantee that the `id` is unique? I guess so, is a `CallInstruction` hashable, that would make me less nervous.

Reply:
`CallInstruction` objects aren't hashable. I've used the `name` attribute instead now.

Reply:
Maybe it is possible to use the function here https://github.com/firedrakeproject/loopy/blob/94f191ce2545de5c959b19844d367c129473e80d/loopy/kernel/instruction.py#L441?

Reply:
Is the kernel `name` attribute not unique?

Reply:
It must be, yes.